server/controllers/user.js

```diff
@@ -0,0 +1,165 @@
+}
+
+module.exports = {
+
+  //create a new user
+
+  create(req, res) {
+    let firstName = req.body.firstName;
+    let lastName = req.body.lastName;
+    let userName = req.body.userName;
+    let email = req.body.email;
+    let password = bcrypt.hashSync(req.body.password, saltRounds);
+
+    if (!firstName || !lastName || !userName || !email || !password) {
+      return res.status(400).json({ message: 'Enter All Required Fields' });
+    } else if (!(/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(req.body.email))) {
+      return res.status(401).json({ message: 'Please Enter A Valid Email Address' });
+    } else {
+      user.create({
+        firstName,
+        lastName,
+        userName,
+        email,
+        password,
+      })
+        .then(user => res.status(201).json(user))
+        .catch(error => res.status(400).send(error));
+    }
+  },
+
+  //login a user
+
+  login(req, res) {
+    user.findOne({
+      where: {
+        email: req.body.email
+      }
+    })
+      .then((user) => {
+        if (!user) {
+          return res.status(403).send({
+            message: 'Invalid user',
+          });
+        }
+        bcrypt.compare(req.body.password, user.password)
+          .then(() => {
+            const token = jwt.sign({ data: user.id }, secretKey, { expiresIn: '24h' });
+            res.status(200).send({
+              message: 'You were successfully logged in',
+              token,
+              expiresIn: '24h'
+            });
+          })
+          .catch(() => {
+            res.status(401).send({
+              message: 'Invalid login credentials',
+            });
+          })
+      })
+      .catch(() => {
+        res.status(401).send({
+          error: 'Invalid login credentials'
+        });
+      });
+  },
+
+  //Find matching instances of user
+
+  list(req, res) {
+    findWithDocuments('findAll')
+      .then(user => res.status(200).send(user))
+      .catch(error => res.status(400).send(error));
+  },
+
+  //Find user
+
+  retrieve(req, res) {
+    findWithDocuments('findById', req.params.userId)
+      .then(user => {
+        if (!user) {
+          return res.status(404).send({
+            message: 'user Not Found',
+          });
+        }
+        return res.status(200).send(user);
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  //Update user attributes
+
+  update(req, res) {
+    findWithDocuments('findById', req.params.userId)
+      .then(user => {
+        if (!user) {
+          return res.status(404).send({
+            message: 'User Not Found',
+          });
+        }
+        user.update({ userName: req.body.userName || user.userName, })
+          .then(() => res.status(200).send(user))
+          .catch((error) => res.status(400).send(error));
+      })
+      .catch((error) => res.status(400).send(error));
+  },
+
+  //delete a user
+
+  destroy(req, res) {
+    findWithDocuments('findById', req.params.userId)
```

Comment:
reolved
` destroy(req, res) {
    user.findById(req.params.userId)
      .then((resp) => {
        if (!resp) {
          return res.status(404).send({
            message: 'User Not Found',
          });
        }
        resp.destroy()
          .then(() => res.status(200).send({ message: 'user deleted' }))
          .catch(error => res.status(400).send(error));
      })
      .catch(error => res.status(400).send(error));
  },`